index.js

```diff
@@ -132,13 +132,47 @@ module.exports = function StateProvider(makeRenderer, rootElement, stateRouterOp
 		return series(stateNames, stateName => renderStateName(parameters, stateName))
 	}
 
+	function allowStateChangeOrRevert(newState, newParameters) {
+		const lastState = lastCompletelyLoadedState.get()
+		if (lastState.name && lastState.name === lastStateStartedActivating.get().name) {
+			// Check allowStateChange for all states that will be destroyed or changed
+			const { change, destroy } = stateChangeLogic(
+				compareStartAndEndStates({
+					original: lastState,
+					destination: {
+						name: newState.name,
+						parameters: newParameters,
+					},
+				}),
+			)
+			const statesNamesToCheck = change?.concat(destroy) ?? destroy ?? []
```

Comment:
I think this works
```suggestion
			const statesNamesToCheck = [ ...change, ...destroy ]
```

Reply:
Should be fine if we can guarantee that change and destroy are always iterables (not undefined). I wasn't certain so I decided to account for that possibility.

Reply:
oh right, I forgot that you can use the object spread operator on null/undefined, but not the array spread operator.  derp.